### ISSUE TITLE
Add overview of PropTypes for each component to storybook

### ISSUE THREAD

**Issue by hborghols** [MyOnlineStore/bricks#390]
### Is your feature request related to a problem? Please describe ⚡️
Currently we don't have any documentation showing the API of our components. To get an overview of which props are available, required and what values they take it's necessary to dive into the source code or rely on typeerrors, both of which work but are not the most user-friendly.

### Describe the solution you'd like ✨
I made a small PoC that uses https://github.com/storybookjs/storybook/tree/next/addons/info and https://github.com/strothj/react-docgen-typescript-loader/ to be able to generate tables of PropTypes for components. You can find this in the [poc-proptables branch](https://github.com/MyOnlineStore/bricks/tree/poc-proptables).

I propose adding a separate `PropsType` or `Proptable` story for each component that uses the info plugin with the `inline` option set to `true` to immediately show a table of props. This also requires adding descriptions with additional information to the type definitions. Adding a story with info settings is as simple as:

```
storiesOf('Modal', module).add(
   'Props table',
   withInfo({ inline: true })(() => {
       return <Modal show={false} size="medium" title="Props table!" />;
   }),
);
```
Adding additional descriptions to the type definition simply uses comments:
```
type PropsType = {
    /** Determines whether the modal is visible or not */
    show: boolean;

    /** Title shown at the top of the modal */
    title: string;

    etc...
```
Descriptions are not necessary for all props but are useful for some, e.g., it's not necessary to explain `show: boolean;` but it may be useful to add some context to `renderFixed?(): JSX.Element`.

### Describe alternatives you've considered 🎡
- Add an info tab to all existing stories. This is easily possible by adding `addDecorator(withInfo)` to the storybook `config.js` file. However, we have quite a few elaborate demo stories which are not suitable for generating a proptable. The plugin generates a table for each element used in the story so it's clearest is the story *only* uses the element we want to document.
- Add a clickable info tab to current stories where suitable and add new ones where necessary. I personally prefer the inline view and it seems like this would still require adding quite a few new stories to be able to show the tables properly. Converting current stories may be possible with the proposed solution anyway.
- There doesn't really seem to be a good alternative way to generate prop tables in storybook. I haven't looked into options outside of storybook yet.

### Additional challenges
- It may be necessary to filter out some of the props for some of the components. This appears to be possible using the docgen-loader but I haven't looked into this in depth yet. 

### Disadvantages:
- The docgen plugin makes the webpack build significantly slower.

### Additional context 🖼
Example table that I generated in the PoC:
![Screenshot 2019-06-03 at 16 25 22](https://user-images.githubusercontent.com/23050817/58809449-3cdf8a80-861c-11e9-9cc1-16e01ecba6ae.png)

**Comment by CarloPalinckx** [comment]
Looks great! Any reason you opted away from always showing the propsTable?

**Comment by hborghols** [comment]
Yeah as it says somewhere in there 😉: we have quite a few elaborate demo stories which are not suitable for generating a proptable. The plugin generates a table for each element used in the story so it's clearest is the story only uses the element we want to document.

So what you get when always showing the propsTable is multiple tables for a story and sometimes tables of, for example, a div with all it's standard props. As you can imagine this is quite a long list and not always useful.
